config/pipelines/reception.yml

```diff
@@ -0,0 +1,62 @@
+default: &default
+  reception:
+    stock_resource:
+      message:
+        lims: &lims Traction
+        key: stock_resource
+        fields:
+          created_at: &timestamp
+            type: :constant
+            value: Time.current
+          updated_at: *timestamp
+          labware_coordinate:
+            type: :model
+            value: requestable.container_material.container.position
+          machine_barcode:
+            type: :model
+            value: requestable.container_material.container.barcode
+          human_barcode:
+            type: :model
+            value: requestable.container_material.container.barcode
+          labware_type:
+            type: :model
+            value: requestable.container_material.container.labware_type
+          stock_resource_id:
+            type: :model
+            value: requestable.container_material.container.id
+          samples:
+            type: :array
+            value: samples_for_stock_resource
+            children:
+              sample_uuid:
+                type: :model
+                value: sample.external_id
+              study_uuid:
+                type: :model
+                value: requestable.external_study_id
+
+    sample:
+      message:
+        lims: *lims
+        key: sample
+        fields:
+          last_updated: *timestamp
+          id_sample_lims:
+            type: :model
+            value: id
+          uuid_sample_lims:
+            type: :model
+            value: external_id
+          name:
+            type: :model
+            value: name
+          common_name:
+            type: :model
+            value: species
+
+
+development: *default
+test: *default
+uat: *default
+training: *default
+production: *default
```

Comment:
What uses this configuration?
How are values like Time.current expanded?

Reply:
This configuration is the definition of the message payload going to the warehouse from Traction.
There is some code in ```app/messages/messages/message.rb``` that evaluates the expression of value before sending.